### ISSUE TITLE
debug_wrapper raised RuntimeError: Cannot call sizes() on tensor with symbolic sizes/strides

### ISSUE THREAD

**Issue by Odin08** [pytorch/pytorch#98849]
### 🐛 Describe the bug

Dropout don't work for simple example with "dynamic=True".

### Error logs

```
Failed to collect metadata on function, produced code may be suboptimal.  Known situations this can occur are inference mode only compilation involving resize_ or prims (!schema.hasAnyAliasInfo() INTERNAL ASSERT FAILED); if your situation looks different please file a bug to PyTorch.
Traceback (most recent call last):
  File "/home/vgonta/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py", line 1674, in aot_wrapper_dedupe
    fw_metadata, _out = run_functionalized_fw_and_collect_metadata(
  File "/home/vgonta/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py", line 606, in inner
    flat_f_outs = f(*flat_f_args)
  File "/home/vgonta/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py", line 2776, in functional_call
    out = Interpreter(mod).run(*args[params_len:], **kwargs)
  File "/home/vgonta/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/fx/interpreter.py", line 136, in run
    self.env[node] = self.run_node(node)
  File "/home/vgonta/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/fx/interpreter.py", line 177, in run_node
    return getattr(self, n.op)(n.target, args, kwargs)
  File "/home/vgonta/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/fx/interpreter.py", line 294, in call_module
    return submod(*args, **kwargs)
  File "/home/vgonta/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/vgonta/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/nn/modules/dropout.py", line 59, in forward
    return F.dropout(input, self.p, self.training, self.inplace)
  File "/home/vgonta/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/nn/functional.py", line 1249, in dropout
    return handle_torch_function(dropout, (input,), input, p=p, training=training, inplace=inplace)
  File "/home/vgonta/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/overrides.py", line 1534, in handle_torch_function
    result = mode.__torch_function__(public_api, types, args, kwargs)
  File "/home/vgonta/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_inductor/overrides.py", line 37, in __torch_function__
    return replacements[func](*args, **kwargs)
  File "/home/vgonta/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_inductor/overrides.py", line 558, in lowmem_dropout
    result = LowmemDropout.apply(input, p)
  File "/home/vgonta/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/autograd/function.py", line 506, in apply
    return super().apply(*args, **kwargs)  # type: ignore[misc]
RuntimeError: Cannot call sizes() on tensor with symbolic sizes/strides

While executing %self_d : [#users=1] = call_module[target=self_d](args = (%self_fc,), kwargs = {})
Original traceback:
  File "/tmp/ipykernel_44196/2526324608.py", line 14, in forward
    x = self.d(x)

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/output_graph.py:670, in OutputGraph.call_user_compiler(self, gm)
    669 else:
--> 670     compiled_fn = compiler_fn(gm, self.fake_example_inputs())
    671 _step_logger()(logging.INFO, f"done compiler function {name}")

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/debug_utils.py:1055, in wrap_backend_debug.<locals>.debug_wrapper(gm, example_inputs, **kwargs)
   1054 else:
-> 1055     compiled_gm = compiler_fn(gm, example_inputs)
   1057 return compiled_gm

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/__init__.py:1390, in _TorchCompileInductorWrapper.__call__(self, model_, inputs_)
   1388 from torch._inductor.compile_fx import compile_fx
-> 1390 return compile_fx(model_, inputs_, config_patches=self.config)

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_inductor/compile_fx.py:401, in compile_fx(model_, example_inputs_, inner_compile, config_patches)
    400     with config.patch(config_patches):
--> 401         return compile_fx(
    402             model_,
    403             example_inputs_,
    404             # need extra layer of patching as backwards is compiled out of scope
    405             inner_compile=config.patch(config_patches)(inner_compile),
    406         )
    408 assert not config._raise_error_for_testing

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_inductor/compile_fx.py:455, in compile_fx(model_, example_inputs_, inner_compile, config_patches)
    450 with overrides.patch_functions():
    451 
    452     # TODO: can add logging before/after the call to create_aot_dispatcher_function
    453     # in torch._functorch/aot_autograd.py::aot_module_simplified::aot_function_simplified::new_func
    454     # once torchdynamo is merged into pytorch
--> 455     return aot_autograd(
    456         fw_compiler=fw_compiler,
    457         bw_compiler=bw_compiler,
    458         decompositions=select_decomp_table(),
    459         partition_fn=functools.partial(
    460             min_cut_rematerialization_partition, compiler="inductor"
    461         ),
    462         keep_inference_input_mutations=True,
    463     )(model_, example_inputs_)

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/backends/common.py:48, in aot_autograd.<locals>.compiler_fn(gm, example_inputs)
     47 with enable_aot_logging():
---> 48     cg = aot_module_simplified(gm, example_inputs, **kwargs)
     49     counters["aot_autograd"]["ok"] += 1

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py:2805, in aot_module_simplified(mod, args, fw_compiler, bw_compiler, partition_fn, decompositions, hasher_type, static_argnums, keep_inference_input_mutations)
   2803 full_args.extend(args)
-> 2805 compiled_fn = create_aot_dispatcher_function(
   2806     functional_call,
   2807     full_args,
   2808     aot_config,
   2809 )
   2811 # TODO: There is something deeply wrong here; compiled_fn running with
   2812 # the boxed calling convention, but aot_module_simplified somehow
   2813 # historically returned a function that was not the boxed calling
   2814 # convention.  This should get fixed...

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/utils.py:163, in dynamo_timed.<locals>.dynamo_timed_inner.<locals>.time_wrapper(*args, **kwargs)
    162 t0 = time.time()
--> 163 r = func(*args, **kwargs)
    164 time_spent = time.time() - t0

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py:2498, in create_aot_dispatcher_function(flat_fn, flat_args, aot_config)
   2496 # You can put more passes here
-> 2498 compiled_fn = compiler_fn(flat_fn, fake_flat_args, aot_config)
   2500 if not hasattr(compiled_fn, "_boxed_call"):

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py:1802, in aot_wrapper_dedupe(flat_fn, flat_args, aot_config, compiler_fn)
   1800     return flat_fn(*add_dupe_args(args))
-> 1802 compiled_fn = compiler_fn(wrapped_flat_fn, deduped_flat_args, aot_config)
   1804 if not hasattr(compiled_fn, "_boxed_call"):

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py:2036, in aot_dispatch_autograd(flat_fn, flat_args, aot_config)
   2035 with enable_python_dispatcher():
-> 2036     _fw_metadata, out = run_functionalized_fw_and_collect_metadata(
   2037         flat_fn,
   2038         # Note: in the non-inference path, we are currently not passing input mutations into the graph directly.
   2039         # This is mainly difficult due to the partitioner, but we are leaving (a bit of) perf on the table.
   2040         keep_input_mutations=False,
   2041     )(
   2042         *flat_args
   2043     )
   2046 # out here corresponds to the set of outputs in the traced forward that should get grad_outputs in the traced backward.
   2047 # It includes outputs of the original forward, *and* any updated inputs due to input mutations.
   2048 # However, it does *not* include any outputs that are aliases of inputs or intermediates, or any metadata-only input mutations.

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py:606, in run_functionalized_fw_and_collect_metadata.<locals>.inner(*flat_args)
    604 try:
    605     # precondition: The passed in function already handles unflattening inputs + flattening outputs
--> 606     flat_f_outs = f(*flat_f_args)
    607 finally:

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py:1800, in aot_wrapper_dedupe.<locals>.wrapped_flat_fn(*args)
   1798 @wraps(flat_fn)
   1799 def wrapped_flat_fn(*args):
-> 1800     return flat_fn(*add_dupe_args(args))

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py:2776, in aot_module_simplified.<locals>.functional_call(*args, **kwargs)
   2775         with torch.autograd.detect_anomaly(check_nan=False):
-> 2776             out = Interpreter(mod).run(*args[params_len:], **kwargs)
   2777 else:

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/fx/interpreter.py:136, in Interpreter.run(self, initial_env, enable_io_processing, *args)
    135 try:
--> 136     self.env[node] = self.run_node(node)
    137 except Exception as e:

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/fx/interpreter.py:177, in Interpreter.run_node(self, n)
    176 assert isinstance(kwargs, dict)
--> 177 return getattr(self, n.op)(n.target, args, kwargs)

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/fx/interpreter.py:294, in Interpreter.call_module(self, target, args, kwargs)
    292 submod = self.fetch_attr(target)
--> 294 return submod(*args, **kwargs)

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/nn/modules/module.py:1501, in Module._call_impl(self, *args, **kwargs)
   1498 if not (self._backward_hooks or self._backward_pre_hooks or self._forward_hooks or self._forward_pre_hooks
   1499         or _global_backward_pre_hooks or _global_backward_hooks
   1500         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1501     return forward_call(*args, **kwargs)
   1502 # Do not call functions when jit is used

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/nn/modules/dropout.py:59, in Dropout.forward(self, input)
     58 def forward(self, input: Tensor) -> Tensor:
---> 59     return F.dropout(input, self.p, self.training, self.inplace)

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/nn/functional.py:1249, in dropout(input, p, training, inplace)
   1248 if has_torch_function_unary(input):
-> 1249     return handle_torch_function(dropout, (input,), input, p=p, training=training, inplace=inplace)
   1250 if p < 0.0 or p > 1.0:

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/overrides.py:1534, in handle_torch_function(public_api, relevant_args, *args, **kwargs)
   1533 with _pop_mode_temporarily() as mode:
-> 1534     result = mode.__torch_function__(public_api, types, args, kwargs)
   1535 if result is not NotImplemented:

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_inductor/overrides.py:37, in AutogradMonkeypatch.__torch_function__(self, func, types, args, kwargs)
     33 if func in replacements and not (
     34     config.fallback_random
     35     and replacements[func] in replacements_using_triton_random
     36 ):
---> 37     return replacements[func](*args, **kwargs)
     38 return func(*args, **kwargs)

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_inductor/overrides.py:558, in lowmem_dropout(input, p, training, inplace)
    557     return input
--> 558 result = LowmemDropout.apply(input, p)
    559 if inplace:

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/autograd/function.py:506, in Function.apply(cls, *args, **kwargs)
    505     args = _functorch.utils.unwrap_dead_wrappers(args)
--> 506     return super().apply(*args, **kwargs)  # type: ignore[misc]
    508 if cls.setup_context == _SingleLevelFunction.setup_context:

RuntimeError: Cannot call sizes() on tensor with symbolic sizes/strides

While executing %self_d : [#users=1] = call_module[target=self_d](args = (%self_fc,), kwargs = {})
Original traceback:
  File "/tmp/ipykernel_44196/2526324608.py", line 14, in forward
    x = self.d(x)


The above exception was the direct cause of the following exception:

BackendCompilerFailed                     Traceback (most recent call last)
Cell In[1], line 23
     20 tensor = torch.randn(5, 10, 16)
     22 model.train()
---> 23 model(tensor)

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/nn/modules/module.py:1501, in Module._call_impl(self, *args, **kwargs)
   1496 # If we don't have any hooks, we want to skip the rest of the logic in
   1497 # this function, and just call forward.
   1498 if not (self._backward_hooks or self._backward_pre_hooks or self._forward_hooks or self._forward_pre_hooks
   1499         or _global_backward_pre_hooks or _global_backward_hooks
   1500         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1501     return forward_call(*args, **kwargs)
   1502 # Do not call functions when jit is used
   1503 full_backward_hooks, non_full_backward_hooks = [], []

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/eval_frame.py:82, in OptimizedModule.forward(self, *args, **kwargs)
     81 def forward(self, *args, **kwargs):
---> 82     return self.dynamo_ctx(self._orig_mod.forward)(*args, **kwargs)

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/eval_frame.py:209, in _TorchDynamoContext.__call__.<locals>._fn(*args, **kwargs)
    207 dynamic_ctx.__enter__()
    208 try:
--> 209     return fn(*args, **kwargs)
    210 finally:
    211     set_eval_frame(prior)

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/eval_frame.py:337, in catch_errors_wrapper.<locals>.catch_errors(frame, cache_size)
    334             return hijacked_callback(frame, cache_size, hooks)
    336 with compile_lock:
--> 337     return callback(frame, cache_size, hooks)

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:404, in convert_frame.<locals>._convert_frame(frame, cache_size, hooks)
    402 counters["frames"]["total"] += 1
    403 try:
--> 404     result = inner_convert(frame, cache_size, hooks)
    405     counters["frames"]["ok"] += 1
    406     return result

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:104, in wrap_convert_context.<locals>._fn(*args, **kwargs)
    102 torch.fx.graph_module._forward_from_src = fx_forward_from_src_skip_result
    103 try:
--> 104     return fn(*args, **kwargs)
    105 finally:
    106     torch._C._set_grad_enabled(prior_grad_mode)

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:262, in convert_frame_assert.<locals>._convert_frame_assert(frame, cache_size, hooks)
    259 global initial_grad_state
    260 initial_grad_state = torch.is_grad_enabled()
--> 262 return _compile(
    263     frame.f_code,
    264     frame.f_globals,
    265     frame.f_locals,
    266     frame.f_builtins,
    267     compiler_fn,
    268     one_graph,
    269     export,
    270     hooks,
    271     frame,
    272 )

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/utils.py:163, in dynamo_timed.<locals>.dynamo_timed_inner.<locals>.time_wrapper(*args, **kwargs)
    161     compilation_metrics[key] = []
    162 t0 = time.time()
--> 163 r = func(*args, **kwargs)
    164 time_spent = time.time() - t0
    165 # print(f"Dynamo timer: key={key}, latency={latency:.2f} sec")

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:324, in _compile(code, globals, locals, builtins, compiler_fn, one_graph, export, hooks, frame)
    322 for attempt in itertools.count():
    323     try:
--> 324         out_code = transform_code_object(code, transform)
    325         orig_code_map[out_code] = code
    326         break

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/bytecode_transformation.py:445, in transform_code_object(code, transformations, safe)
    442 instructions = cleaned_instructions(code, safe)
    443 propagate_line_nums(instructions)
--> 445 transformations(instructions, code_options)
    446 return clean_and_assemble_instructions(instructions, keys, code_options)[1]

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:311, in _compile.<locals>.transform(instructions, code_options)
    298 nonlocal output
    299 tracer = InstructionTranslator(
    300     instructions,
    301     code,
   (...)
    309     mutated_closure_cell_contents,
    310 )
--> 311 tracer.run()
    312 output = tracer.output
    313 assert output is not None

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py:1726, in InstructionTranslator.run(self)
   1724 def run(self):
   1725     _step_logger()(logging.INFO, f"torchdynamo start tracing {self.f_code.co_name}")
-> 1726     super().run()

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py:576, in InstructionTranslatorBase.run(self)
    571 try:
    572     self.output.push_tx(self)
    573     while (
    574         self.instruction_pointer is not None
    575         and not self.output.should_exit
--> 576         and self.step()
    577     ):
    578         pass
    579 except BackendCompilerFailed:

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py:540, in InstructionTranslatorBase.step(self)
    538     if not hasattr(self, inst.opname):
    539         unimplemented(f"missing: {inst.opname}")
--> 540     getattr(self, inst.opname)(inst)
    542     return inst.opname != "RETURN_VALUE"
    543 except BackendCompilerFailed:

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py:1792, in InstructionTranslator.RETURN_VALUE(self, inst)
   1787 _step_logger()(
   1788     logging.INFO,
   1789     f"torchdynamo done tracing {self.f_code.co_name} (RETURN_VALUE)",
   1790 )
   1791 log.debug("RETURN_VALUE triggered compile")
-> 1792 self.output.compile_subgraph(
   1793     self, reason=GraphCompileReason("return_value", [self.frame_summary()])
   1794 )
   1795 self.output.add_output_instructions([create_instruction("RETURN_VALUE")])

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/output_graph.py:517, in OutputGraph.compile_subgraph(self, tx, partial_convert, reason)
    503     self.add_output_instructions(random_calls_instructions)
    505 if (
    506     stack_values
    507     and all(
   (...)
    514 
    515     # optimization to generate better code in a common case
    516     self.add_output_instructions(
--> 517         self.compile_and_call_fx_graph(tx, list(reversed(stack_values)), root)
    518         + [create_instruction("UNPACK_SEQUENCE", len(stack_values))]
    519     )
    520 else:
    521     graph_output_var = self.new_var("graph_out")

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/output_graph.py:588, in OutputGraph.compile_and_call_fx_graph(self, tx, rv, root)
    586 assert_no_fake_params_or_buffers(gm)
    587 with tracing(self.tracing_context):
--> 588     compiled_fn = self.call_user_compiler(gm)
    589 compiled_fn = disable(compiled_fn)
    591 counters["stats"]["unique_graphs"] += 1

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/utils.py:163, in dynamo_timed.<locals>.dynamo_timed_inner.<locals>.time_wrapper(*args, **kwargs)
    161     compilation_metrics[key] = []
    162 t0 = time.time()
--> 163 r = func(*args, **kwargs)
    164 time_spent = time.time() - t0
    165 # print(f"Dynamo timer: key={key}, latency={latency:.2f} sec")

File ~/.pyenv/versions/3.9.13/envs/speech-synthesis/lib/python3.9/site-packages/torch/_dynamo/output_graph.py:675, in OutputGraph.call_user_compiler(self, gm)
    673 except Exception as e:
    674     compiled_fn = gm.forward
--> 675     raise BackendCompilerFailed(self.compiler_fn, e) from e
    676 return compiled_fn

BackendCompilerFailed: debug_wrapper raised RuntimeError: Cannot call sizes() on tensor with symbolic sizes/strides

While executing %self_d : [#users=1] = call_module[target=self_d](args = (%self_fc,), kwargs = {})
Original traceback:
  File "/tmp/ipykernel_44196/2526324608.py", line 14, in forward
    x = self.d(x)


Set torch._dynamo.config.verbose=True for more information


You can suppress this exception and fall back to eager by setting:
    torch._dynamo.config.suppress_errors = True
```

### Minified repro

```
import torch
from torch import nn
class D(nn.Module):
    def __init__(self):
        super().__init__()
        self.fc = nn.Linear(16, 16)
        self.d = nn.Dropout(p=0.5)
    def forward(self, x):
        x = self.fc(x)
        x = self.d(x)
        return x

model = D()
model = torch.compile(model, dynamic=True)
tensor = torch.randn(5, 10, 16)
model.train()
model(tensor)
```

### Versions

Collecting environment information...
PyTorch version: 2.0.0+cu118
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.2 LTS (x86_64)
GCC version: (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Clang version: Could not collect
CMake version: version 3.26.0
Libc version: glibc-2.35

Python version: 3.9.13 (main, Sep  6 2022, 01:37:16)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.19.0-38-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 4090
Nvidia driver version: 530.30.02
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Address sizes:                   46 bits physical, 48 bits virtual
Byte Order:                      Little Endian
CPU(s):                          32
On-line CPU(s) list:             0-31
Vendor ID:                       GenuineIntel
Model name:                      13th Gen Intel(R) Core(TM) i9-13900K
CPU family:                      6
Model:                           183
Thread(s) per core:              2
Core(s) per socket:              24
Socket(s):                       1
Stepping:                        1
CPU max MHz:                     6000,0000
CPU min MHz:                     800,0000
BogoMIPS:                        5990.40
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid rdseed adx smap clflushopt clwb intel_pt sha_ni xsaveopt xsavec xgetbv1 xsaves split_lock_detect avx_vnni dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp hwp_pkg_req hfi umip pku ospke waitpkg gfni vaes vpclmulqdq tme rdpid movdiri movdir64b fsrm md_clear serialize pconfig arch_lbr ibt flush_l1d arch_capabilities
Virtualization:                  VT-x
L1d cache:                       896 KiB (24 instances)
L1i cache:                       1,3 MiB (24 instances)
L2 cache:                        32 MiB (12 instances)
L3 cache:                        36 MiB (1 instance)
NUMA node(s):                    1
NUMA node0 CPU(s):               0-31
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Retbleed:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected

Versions of relevant libraries:
[pip3] numpy==1.23.5
[pip3] torch==2.0.0+cu118
[pip3] torchaudio==2.0.1+cu118
[pip3] triton==2.0.0
[conda] Could not collect

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh

**Comment by ngimel** [comment]
Can you please provide runnable example demonstrating the problem?
@nkaretnikov I thought you enabled dropout for dynamic shapes?
